app/workers/deliver_email_worker.rb

```diff
@@ -14,8 +14,19 @@ def self.queue_for_priority(priority)
   sidekiq_options retry: 3, queue: queue_for_priority(:low)
 
   def perform(email_id)
+    start = Time.now.utc
+
     email = Email.find(email_id)
     DeliverEmail.call(email: email)
+
+    duration = (start - Time.now.utc) * 1000.0
+
+    sleep(
```

Comment:
Feels to me like we could have another method in the `RateSleeper` class which does the `sleep` for us and does the check @kevindew mentioned on `sleep(0)`.

Reply:
Maybe even provide something similar to what @tuzz is suggesting and have a helper method which does all the above and accepts a block to execute.

```ruby
RateSleeper.run do
  email = Email.find(email_id)
  DeliverEmail.call(email: email)
end
```